agent/acs/handler/payload_handler.go

```diff
@@ -274,7 +286,7 @@ func (payloadHandler *payloadRequestHandler) handleUnrecognizedTask(task *ecsacs
 	}
 
 	// Only need to stop the task; it brings down the containers too.
-	eventhandler.AddTaskEvent(api.TaskStateChange{
+	payloadHandler.taskHandler.AddTaskEvent(api.TaskStateChange{
```

Comment:
It's mostly what I mentioned earlier. Payload handler needs to generate an event, which is processed by the event handler. Task Engine has no role in this interaction. Just because Task Engine can also send events to the event handler doesn't mean that, that should be the interface used for sending events to the event stream. Does that make sense?